shopfloor_reception/services/reception.py

```diff
@@ -770,6 +770,10 @@ def _align_product_uom_qties(self, move):
             for line in lines:
                 line.product_uom_qty = line.qty_done + remaining_todo
 
+    def _before_response_for_set_quantity(self, picking, line):
```

Comment:
the name of this method seems wrong... Is not "before" is actually replacing the direct call to the other method :thinking:

Reply:
It actually inserts a new front end state before the set quantity state if needed. Names can always be improved...

Reply:
why don't you call it then maybe _response_after_scan_line__assign_user ?

Reply:
well, it is also called atfer `set_lot_confirm_action` so I don't think it it really fits.

Reply:
TBH I'd prefer something more explicit. The main issue here is that we are trying to change the behavior of something that does not take into account the possibility to handle dimensions.
We could assume - that being a WMS app - we can loosely support such extension.

What if we simply offer an explicit hook here (and everywhere else needed)?
Eg: 
```
       if product.tracking not in ("lot", "serial") or (line.lot_id or line.lot_name):
            if self._requires_packaging_dimension(...):
                return self._response_for_set_packaging_dimension(...)
            return self._response_for_set_quantity(picking, line)

[....]

def _response_for_set_packaging_dimension(self, .....):
    raise NotImplementedError()
```

WDYT?

Reply:
I am not sure why this is better. It adds more coupling between module when not needed.

What is being done here is inserting a new backend state in a scenario before an existing one. When that new state has achieved its purpose (multiple conversation/interaction between the frontend and backend) the scenario will proceed as initially intended.

Now I still agree that the naming is not good. Because the insertion should not be done before generating the response for `set_quantity` (which is called for the first and subsequent access to the state) but before entering the state `set_quantity`.

So in that light I refactored the code renaming the function to `_before_state__set_quantity`.
It does not add coupling and can be used by other modules.